test/start.test.js

```diff
@@ -483,6 +483,49 @@ test('crash on unhandled rejection', t => {
   })
 })
 
+test('should start the server with inspect options and the defalut port is 9320', t => {
+  t.plan(4)
+
+  const start = proxyquire('../start', {
+    inspector: {
+      open (p) {
+        t.strictEqual(p, 9320)
+        t.pass('inspect open called')
+      }
+    }
+  })
+  const argv = ['--d', './examples/plugin.js']
+  start.start(argv, (err, fastify) => {
+    t.error(err)
+
+    fastify.close(() => {
+      t.pass('server closed')
+    })
+  })
+})
+
+test('should start the server with inspect options and use the exactly port', t => {
+  t.plan(4)
+
+  const port = getPort()
+  const start = proxyquire('../start', {
+    inspector: {
+      open (p) {
+        t.strictEqual(p, Number(port))
+        t.pass('inspect open called')
+      }
+    }
+  })
+  const argv = ['--d', '--debug-port', port, './examples/plugin.js']
+  start.start(argv, (err, fastify) => {
+    t.error(err)
+
+    fastify.close(() => {
+      t.pass('server closed')
+    })
+  })
+})
```

Comment:
You'll need to skip those tests if Node < 8.

Reply:
@mcollina It's a draft, and I am working on this